setup.md

```diff
@@ -174,44 +174,206 @@ before pushing them to GitHub.
+  ~~~
+  {: .language-bash}
+
+ If you're not sure whether you have brew installed, type
+ 
+  ~~~
+  brew help
+  ~~~
+  {: .language-bash}
+
+  If you have brew installed, you should see something like:
+
+  ~~~
+  Example usage:
+  brew search [TEXT|/REGEX/]
+  brew info [FORMULA...]
+  brew install FORMULA...
+  brew update
+  brew upgrade [FORMULA...]
+  brew uninstall FORMULA...
+  brew list [FORMULA...]
+
+Troubleshooting:
+  brew config
+  brew doctor
+  brew install --verbose --debug FORMULA
+
+Contributing:
+  brew create [URL [--no-fetch]]
+  brew edit [FORMULA...]
+
+Further help:
+  brew commands
+  brew help [COMMAND]
+  man brew
+  https://docs.brew.sh
+  ~~~
+  {: .output}
+ 
+
+2. **Ruby** -- `brew install ruby`
+
+3. **bundler** -- `gem install bundler --user-install`
+
+4. Go to the section [For Everyone](#for-everyone)
+
+### Linux (Ubuntu)
+
+1.  **[Ruby](https://www.ruby-lang.org/en/downloads/)** and other dependencies.
+
+    You will need to have the following packages installed (some might already
+    be on your system):
+    
     ~~~
-    rbenv install 2.5.3
+    sudo apt-get install ruby ruby-dev build-essential libxml2-dev
     ~~~
     {: .language-bash}
-
-    And then instructing `rbenv` to use it in your lesson development process by executing the
-    following command from your lesson directory:
+    
+2. **[bundler](https://bundler.io/)**
 
     ~~~
-    rbenv local 2.5.3
+    gem install bundler --user-install
     ~~~
     {: .language-bash}
+    
+    `gem` is the package management framework for Ruby. It was installed as part
+    of Ruby in the step above. `bundler` is also a package manager but at the
+    scale of a project instead of being system-wide. It makes it easier to
+    manage dependencies.
 
-    To install `rbenv`, please use [rbenv-installer](https://github.com/rbenv/rbenv-installer).
 
-    **Windows**: Please use [RubyInstaller][ruby-installer] to install Ruby on Windows.
+### For Everyone
 
-    Upon installing Ruby, check its version by executing
+1. **The GitHub Pages Ruby Gem**
 
+    Make sure there is a `Gemfile` at the root of your lesson repository. This
+    file should only contain:
+    
+    ~~~
+    source 'https://rubygems.org'
+    gem 'github-pages', group: :jekyll_plugins
     ~~~
-    ruby --version
+    
+    If you don't have it, create it and the two lines above to it.
+    
+    At the root of your repository type
+    
+    ~~~
+    bundle update
+    ~~~
+    {: .language-bash}
+    
+    If you haven't used `bundler` yet for your project, this command will
+    install all the needed dependencies. Otherwise, it will update them to match
+    the current versions used by GitHub Pages.
+    
+4. **Generate the lesson**
+
+    At this stage, you should be able to do:
+    
+    ~~~
+    make serve
+    ~~~
+    {: .language-bash}
+    
+    and see your generated lesson.
```

Comment:
I think this section should be overall more detailed eg:

------

Now you are ready to run jekyll to build your website and run a local server. To do this run:

~~~
make serve
~~~

There will be several lines of output after this. If there were errors or warnings you can use them to fix your lesson and run again if ti it failed.  If it was successful the last few lines will include `Server address: http://217.0.0.1:4000` and `Server running... press ctrl-c to stop.` You can see your rendered site by pointing your browser to the address shown.